third_party/terraform/resources/resource_sql_database_instance.go.erb

```diff
@@ -664,22 +664,30 @@ func resourceSqlDatabaseInstanceCreate(d *schema.ResourceData, meta interface{})
 		return fmt.Errorf("Error setting name: %s", err)
 	}
 
-	// Before we create the instance, check if at least 1 service connection exists for private SQL Instances.
+	// SQL Instances that fail to create are expensive- see https://github.com/hashicorp/terraform-provider-google/issues/7154
+	// We can fail fast to stop instance names from getting reserved.
 	network := d.Get("settings.0.ip_configuration.0.private_network").(string)
 	if network != "" {
-		// Borrow some of the functions from resource_service_networking_connection.go
+		log.Printf("[DEBUG] checking network %q for at least one service networking connection", network)
+		// This call requires projects.get permissions, which may not have been granted to the Terraform actor,
+		// particularly in shared VPC setups. Most will! But it's not strictly required.
 		serviceNetworkingNetworkName, err := retrieveServiceNetworkingNetworkName(d, config, network, userAgent)
 		if err != nil {
-			return err
-		}
-		response, err := config.NewServiceNetworkingClient(userAgent).Services.Connections.List("services/servicenetworking.googleapis.com").
-			Network(serviceNetworkingNetworkName).Do()
-		if err != nil {
-			// It is possible that the identity creating the SQL Instance might not have permissions to call servicenetworking.services.connections.list
-			log.Printf("[WARNING] Failed to list Service Networking of the project. Skipping Service Networking check.")
+			var gerr *googleapi.Error
```

Comment:
```suggestion
			
			// googleapi.Error is a permission error for the precheck
			var gerr *googleapi.Error
```

Reply:
This isn't entirely true! googleapi errors are effectively any error coming from the API client (assuming the base request is successful, otherwise it will be another type). We could filter on permissions errors, but it isn't worth doing so imo (the more fine-grained the filtering, the easier it is to get wrong)